wd_util.c

```diff
@@ -2173,16 +2173,17 @@ int wd_get_lib_file_path(char *lib_file, char *lib_path, bool is_dir)
 	}
 
 	if (is_dir) {
-		(void)snprintf(lib_path, PATH_MAX, "%s", file_path);
-		return 0;
+		len = snprintf(lib_path, PATH_MAX, "%s/%s", file_path, WD_DRV_LIB_DIR);
+		if (len < 0)
+			return -WD_EINVAL;
+	} else {
+		len = snprintf(lib_path, PATH_MAX, "%s/%s/%s", file_path, WD_DRV_LIB_DIR, lib_file);
```

Comment:
Snprintf outliers are only greater than PATH_MAX. Negative values ​​will not be returned.

Reply:
fixed